### PR TITLE
initial changes for 1.13 support

### PR DESCRIPTION
**Pull request by kristofgyuracz**
### What's in this PR?

Changes needed to support Istio version 1.13.

### Checklist

- [x] Implementation tested
- [x] Error handling code meets the [guideline](https://github.com/banzaicloud/pipeline/blob/master/docs/error-handling-guide.md)
- [x] Logging code meets the guideline
- [x] User guide and development docs updated (if needed)

### CODE REVIEW

File: config/manager/kustomization.yaml [hunk 1]
```diff
@@ -4,4 +4,5 @@ apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
 images:
 - name: controller
-  newName: ghcr.io/banzaicloud/istio-operator
+  newName: kristofgyuracz/istio-operator
+  newTag: v2.12.0-rc.2
```
Comment (Laci21) [inline config/manager/kustomization.yaml]: Let's not commit these.

File: internal/assets/manifests/istio-discovery/templates/telemetryv2_1.13.yaml [hunk 2]
```diff
@@ -0,0 +1,601 @@
+{{- if and (eq .Values.global.mode "ACTIVE") .Values.telemetry.enabled .Values.telemetry.v2.enabled }}
```
Comment (Laci21) [inline internal/assets/manifests/istio-discovery/templates/telemetryv2_1.13.yaml:1]: Shouldn't we delete the 1.11 yaml file?

File: scripts/update-istio-dependencies.sh [hunk 3]
```diff
@@ -17,7 +17,7 @@ echo "clone istio api repository"
 git clone -q -b ${version} https://github.com/istio/api
 
 echo "copy dependencies"
-cp -a api/mesh api/networking .
+cp -a api/mesh api/networking api/type .
```
Comment (Laci21) [inline scripts/update-istio-dependencies.sh:20]: what's this `api/type`?

Comment (kristofgyuracz) [reply scripts/update-istio-dependencies.sh:20]: needed for istio deps generation

File: internal/assets/manifests/istio-discovery/templates/telemetryv2_1.13.yaml (hunk 2 above)
Comment (kristofgyuracz) [reply internal/assets/manifests/istio-discovery/templates/telemetryv2_1.13.yaml:1]: done

File: config/samples/servicemesh_v1alpha1_istiomeshgateway.yaml [hunk 4]
```diff
@@ -6,11 +6,11 @@ spec:
...
       labels:
-        app: istio-meshexpansion-gateway
-        gateway-name: istio-meshexpansion-gateway-cp-v19x
+        app: imgw-sample
+        gateway-name: imgw-sample
         gateway-type: ingress
-        istio: meshexpansiongateway
-        istio.io/rev: cp-v19x.istio-system
+        istio: imgw-sample
+        istio.io/rev: cp-v113x.istio-system
```
Comment (Laci21) [inline config/samples/servicemesh_v1alpha1_istiomeshgateway.yaml]: Shouldn't this be
```suggestion
        istio.io/rev: icp-v113x-sample.istio-system
```

File: deploy/charts/istio-operator/Chart.yaml [hunk 5]
```diff
@@ -9,7 +9,7 @@ sources:
 icon: https://istio.io/latest/img/istio-whitelogo-bluebackground-framed.svg
 
 # Based on support status of Istio releases: https://istio.io/latest/docs/releases/supported-releases/#support-status-of-istio-releases
-kubeVersion: ">= 1.19.0-0 < 1.23.0-0"
+kubeVersion: ">= 1.20.0-0 <= 1.23.0-0"
 
-version: 2.0.8
-appVersion: "v2.12.0"
+version: 2.0.9
+appVersion: "v2.13.0"
```
Comment (Laci21) [inline deploy/charts/istio-operator/Chart.yaml]: When merged, I'd suggest to release a dev or rc release first to test, so sth like:
```suggestion
appVersion: "v2.13.0-rc.0"
```

Same for other places where this version number is used.

**Review by Laci21 — COMMENTED** [review]
LGTM, added a few nits and here are a few general comments:

- Just to confirm, there were no relevant helm chart changes that we didn't already have or make sense to port, right? (except for the CRD changes)
- Please bump to Istio 1.13.3 when you have a chance
- Please use the new cluster registry API location (https://github.com/banzaicloud/istio-operator/pull/846)
- It'd be nice to bump to Go 1.18 (refs that might help: https://github.com/cisco-open/cluster-registry-controller/pull/2, https://github.com/cisco-open/cluster-registry-controller/pull/3/commits/e2d4f8880fa5791e9af8c64dfba0d598a7dfc38f)

File: config/samples/servicemesh_v1alpha1_istiomeshgateway.yaml [hunk 6]
```diff
@@ -6,11 +6,11 @@ spec:
...
       labels:
-        app: istio-meshexpansion-gateway
-        gateway-name: istio-meshexpansion-gateway-cp-v19x
+        app: imgw-sample
+        gateway-name: imgw-sample
         gateway-type: ingress
-        istio: meshexpansiongateway
-        istio.io/rev: cp-v19x.istio-system
+        istio: imgw-sample
+        istio.io/rev: icp-v113x.istio-system
```
Comment (Laci21) [inline config/samples/servicemesh_v1alpha1_istiomeshgateway.yaml]:
```suggestion
        istio.io/rev: icp-v113x-sample.istio-system
```

**Review by Laci21 — APPROVED** [review]
Thank you! Nice job, @kristofgyuracz!

**Pull request merged**